core/src/main/java/hudson/model/UpdateCenter.java

```diff
@@ -1207,7 +1209,11 @@ public UpdateCenterConfiguration() {
          * @throws IOException if a connection can't be established
          */
         public void checkConnection(ConnectionCheckJob job, String connectionCheckUrl) throws IOException {
-            testConnection(new URL(connectionCheckUrl));
+            try {
+                testConnection(new URI(connectionCheckUrl).toURL());
+            } catch (URISyntaxException e) {
```

Comment:
do not swallow the exception here.   this already throws `MalformedURLException` (via IOException declaration)

Reply:
added
```
MalformedURLException mex = new MalformedURLException(e.getMessage());
            mex.initCause(e);
            throw mex;
```